pymongo/change_stream.py

```diff
@@ -283,12 +298,15 @@ def try_next(self):
         # one resume attempt.
         try:
             change = self._cursor._try_next(True)
-        except ConnectionFailure:
+        except (ConnectionFailure, CursorNotFound):
```

Comment:
This addition makes it so that we resume on `CursorNotFound` exceptions. This is a deviation from the current spec, but I believe this to be the appropriate behavior. 
Note that this deviation from the spec allows a lot of PyMongo's tests (non-spec) to pass. Without this change, all tests that use `kill_cursors` to force a change stream resumption start failing.

Reply:
With https://jira.mongodb.org/browse/PYTHON-2302 having been filed, we can proceed with merging this as-is and come back to deal with `CursorNotFound` errors at a later time when the spec has been finalized.

Reply:
Nice find! For the record, I think we should keep resuming on CursorNotFound.

Reply:
Agreed. Rachelle also agreed with this. We can break this in 4.0 but not before.